src/main/java/org/broadinstitute/hellbender/engine/ReadsDataSource.java

```diff
@@ -146,22 +152,52 @@ public ReadsDataSource( final List<File> samFiles, SamReaderFactory customSamRea
      *
      * @param intervals Our next full traversal will return only reads overlapping these intervals
      */
-    public void setIntervalsForTraversal( final List<SimpleInterval> intervals ){
-        // Treat null and empty interval lists the same
-        this.intervals = (intervals != null && ! intervals.isEmpty()) ? intervals : null;
+    public void setTraversalBounds( final List<SimpleInterval> intervals ) {
```

Comment:
Why do we need method?  Is there any non-test that should be calling this method?  It seems like including this may lead to confusion when people don't realize they're forgetting about unmapped reads.